src/main/java/com/github/princesslana/slothbot/MessageCounter.java

```diff
@@ -27,18 +38,32 @@
   private final AtomicReference<ImmutableList<ImmutableBucket>> pastBuckets =
       new AtomicReference<>(ImmutableList.of());
 
+  private final ConcurrentHashMap<String, Boolean> botCounterConfigs = new ConcurrentHashMap<>();
+
   private final ScheduledExecutorService executor;
+  private final Path savePath;
 
-  public MessageCounter(ScheduledExecutorService executor) {
+  public MessageCounter(ScheduledExecutorService executor, Path savePath) {
     this.executor = executor;
+    this.savePath = savePath;
   }
 
   private void onGatewayPayload(String payload) {
     Discord.ifEvent(JsonF.parse(payload), "MESSAGE_CREATE", this::onMessageCreate);
   }
 
   private void onMessageCreate(JsonF d) {
-    d.get("channel_id").asString().ifPresent(currentBucket.get()::increment);
+    d.get("guild_id")
+        .asString()
+        .ifPresent(
+            guildId -> {
+              if (getBotCounterConfig(guildId)) {
+                d.get("channel_id").asString().ifPresent(currentBucket.get()::increment);
+              } else if (!d.get("webhook_id").asString().isPresent()
```

Comment:
Could probably be refactored out to a method like `isBotMessage`, or even just to a variable named that. Makes it clearer what this is actually checking for.